report/report.go

```
@@ -0,0 +1,53 @@
+// package main contains function that calls the service N = 1000 times for each of the strings of the length 2,4,6,8,10.
+// Its calculate the percent of balanced strings for each length and print the results to stdout.
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"github.com/GoRunMaxim/Balanced-Parentheses/parentheses"
+)
+
+func main() {
+	strings := []struct {
+		length int
+	}{
+		{length: 2},
+		{length: 4},
+		{length: 6},
+		{length: 8},
+		{length: 10},
+	}
+	for _, v := range strings {
+		var balancedPercent int
+		for i := 0; i < 1000; i++ {
+			resp, err := http.Get("http://localhost:8080/generate?n=" + fmt.Sprint(v.length))
+			if err != nil {
+				log.Fatal("Request error:" + err.Error())
+			}
+			defer func() {
+				err = resp.Body.Close()
+				if err != nil {
+					log.Fatalf("Cannot close reponce body: %v", err.Error())
+				}
+			}()
```

This might sound a little surprising, but that is one of the cases where you might ignore the error, there's nothing you can really do if it occurs.

---

Golinter told me that its bad practise to ignore error, so at least we need to log it =)

---

Yes, but the linter told me that its bad idea not to check error

---

In general, you should never ignore errors, but rather handle them. As I mentioned, this is one of these rare cases when you can ignore it. Here's why, you close the body to free resources, but if you get an error, you cannot handle it anyhow, but if you report the error, you still can't do anything about it. And it does not make sense to terminate the program. Do you can safely ignore it.